lib/Basic/Subprocess.cpp

```diff
@@ -102,6 +102,9 @@ static int posix_spawn_file_actions_addchdir_polyfill(posix_spawn_file_actions_t
   //  - FreeBSD 13.1 (May 2022)
   //  - Android 14 (October 2023)
   return posix_spawn_file_actions_addchdir_np((posix_spawn_file_actions_t *)file_actions, path);
+#elif defined(__ANDROID__)
```

Comment:
Actually, would you mind moving this up below the glibc case at the top, and write like:

```c
#elif defined(__ANDROID__) && __ANDROID_API__ < 34
```

...to keep the "missing" cases consolidated at the top?

Reply:
> Actually, would you mind moving this up below the glibc case at the top

Done!

Reply:
You missed moving it up. I'd also prefer if you combined it with the glibc and OpenBSD cases, since all return the same error.

You can also remove checking the API level is greater than 34 for the affirmative case above.

Reply:
> I'd also prefer if you combined it with the glibc and OpenBSD cases, since all return the same error.

You mean like this? It seems rather tortuous… 

```
#if (defined(__GLIBC__) && !__GLIBC_PREREQ(2, 29)) || (defined(__OpenBSD__)) || (defined(__ANDROID__) && __ANDROID_API__ < 34)
  return ENOSYS;
```

Reply:
Yes, either way, Jake asked to move it up above. I prefer it combined like that too.